### PR TITLE
Finally fix the image-based SD tests

### PR DESCRIPTION
**Pull request by anton-l**
Solving both the datasets caching and the GPU inconsistency issues

### CODE REVIEW

File: src/diffusers/testing_utils.py [hunk 1]
```diff
@@ -59,3 +63,32 @@ def slow(test_case):
 
     """
     return unittest.skipUnless(_run_slow_tests, "test is slow")(test_case)
+
+
+def load_image(image: Union[str, PIL.Image.Image]) -> PIL.Image.Image:
```
Comment (patrickvonplaten) [inline src/diffusers/testing_utils.py:68]: nice good idea for testing

**Review by pcuenca — APPROVED** [review]
Looks great!

File: tests/test_pipelines.py [hunk 2]
```diff
@@ -1139,23 +1138,48 @@ def test_stable_diffusion_memory_chunking(self):
...
+        image = output.images[0]
+
+        expected_array = np.array(output_image, dtype=np.int32)
+        sampled_array = np.array(image, dtype=np.int32)
 
-        init_image = ds["input"]["image"][0].resize((768, 512))
-        output_image = ds["output"]["image"][0].resize((768, 512))
+        assert sampled_array.shape == (512, 512, 3)
+        # using the mean absolute error due to slightly inconsistent outputs across different GPUs
+        assert np.mean(np.abs(sampled_array - expected_array)) < 2.0
```
Comment (patrickvonplaten) [inline tests/test_pipelines.py:1168]: 2.0 seems to large to me here as discussed. Can we see how big the difference is for pure numpy arrays without **any** optimization? 

Will run:

```python
model_id = "CompVis/stable-diffusion-v1-4"
pipe = StableDiffusionPipeline.from_pretrained(
    model_id,
    use_auth_token=True,
)
pipe.to(torch_device)

prompt = "A fantasy landscape, trending on artstation"
generator = torch.Generator(device=torch_device).manual_seed(0)
output = pipe(prompt=prompt, guidance_scale=7.5, generator=generator, output_type="np")
```

from current master @anton-l could you do the same and then we compare numpy results across `vorace`, `colab` and `brutasse`

Comment (patrickvonplaten) [reply tests/test_pipelines.py:1168]: Note that I just merged @patil-suraj improved attention slicing when disabled, so let's try to include this PR when running the tests

File: tests/test_pipelines.py [hunk 3]
```diff
@@ -1218,36 +1237,28 @@ def test_stable_diffusion_img2img_pipeline_k_lms(self):
...
 
-        expected_array = np.array(output_image) / 255.0
-        sampled_array = np.array(image) / 255.0
+        expected_array = np.array(output_image, dtype=np.int32)
+        sampled_array = np.array(image, dtype=np.int32)
 
         assert sampled_array.shape == (512, 768, 3)
-        assert np.max(np.abs(sampled_array - expected_array)) < 1e-4
+        # using the mean absolute error due to slightly inconsistent outputs across different GPUs
+        assert np.mean(np.abs(sampled_array - expected_array)) < 2.0
```
Comment (patrickvonplaten) [inline tests/test_pipelines.py:1245]: As discussed this can be updated now to much lower tolerance :-)

**Review by patil-suraj — APPROVED** [review]
LGTM! Same comments as Patrick's

**Comment by anton-l** [comment]
Without `autocast` we've got much more consistent results now, even without using numpy arrays for the references. cc @patrickvonplaten

**Pull request merged**